src/Microsoft.Health.Fhir.Shared.Web/Program.cs

```diff
@@ -26,9 +24,8 @@ public static void Main(string[] args)
                     var keyVaultEndpoint = builtConfig["KeyVault:Endpoint"];
                     if (!string.IsNullOrEmpty(keyVaultEndpoint))
                     {
-                        var azureServiceTokenProvider = new AzureServiceTokenProvider();
-                        var keyVaultClient = new KeyVaultClient(new KeyVaultClient.AuthenticationCallback(azureServiceTokenProvider.KeyVaultTokenCallback));
-                        builder.AddAzureKeyVault(keyVaultEndpoint, keyVaultClient, new DefaultKeyVaultSecretManager());
+                        var credential = new DefaultAzureCredential();
+                        builder.AddAzureKeyVault(new System.Uri(keyVaultEndpoint), credential);
```

Comment:
Used [this ](https://stackoverflow.com/questions/62503022/net-core-key-vault-configuration-using-azure-security-keyvault-secrets
)as reference to move to new package Azure.Extensions.AspNetCore.Configuration.Secrets